src/main/java/com/fasterxml/jackson/databind/deser/impl/PropertyValueBuffer.java

```diff
@@ -335,7 +336,12 @@ public Object handleIdValue(final DeserializationContext ctxt, Object bean) thro
                 // also: may need to set a property value as well
                 SettableBeanProperty idProp = _objectIdReader.idProperty;
                 if (idProp != null) {
-                    return idProp.setAndReturn(bean, _idValue);
+                    // [databind#5328] Records do not have setters, skip...to set id value
+                    if (ClassUtil.isRecordType(bean.getClass())) {
```

Comment:
I'll have to think about this: ideally, `setAndReturn()` would not be called because id was passed via Constructor and no check was needed here.

Or maybe check could be done that if `idProp` is a `CreatorProperty`, no call would be made?

This is related to ensuring that we also do not try forcibly setting `Field` in regular POJO case, if id was already passed via Creator (see my note on test class).